src/components/Versions.vue

```diff
@@ -16,27 +16,43 @@
             </div>
         </div>
     </div>
+    <div v-else>
+        <loading-spinner />
+    </div>
 </template>
+
 <script lang="ts">
     import { useStore } from "../store/store";
     import AuthenticationManagement from "../api/AuthenticationManagement";
     import CourseManagement from "../api/CourseManagement";
     import UserManagement from "../api/UserManagement";
     import HyperledgerManagement from "../api/HyperledgerManagement";
     import HyperledgerCourseManagement from "../api/HyperledgerCourseManagement";
+    import { checkPrivilege } from "@/use/PermissionHelper";
+    import { Role } from "@/entities/Role";
+    import { ref, onBeforeMount } from "vue";
+    import LoadingSpinner from "@/components/loading/Spinner.vue";
 
     export default {
-        emits: ["versions"],
-        async setup(props: any, { emit }: any) {
-            let store = useStore();
+        components: {
+            LoadingSpinner,
+        },
+
+        async beforeRouteEnter(_from: any, _to: any, next: any) {
+            const response = await checkPrivilege(Role.LECTURER, Role.STUDENT, Role.ADMIN);
 
```

Comment:
Why are we required to log in, to view the versions? This is not even a route in the first place, so this code does not have any function
```suggestion

```